pr-checks/checks/MilestoneCheck.ts

```diff
@@ -24,7 +24,17 @@ export class MilestoneCheck extends Check {
 			async (ctx) => {
 				const pr = context.payload.pull_request as EventPayloads.WebhookPayloadPullRequestPullRequest
 
-				if (pr && pr.milestone) {
+				if (!pr) {
+					return this.failure(ctx, '')
+				}
+
+				// We ignore PRs whose base branch is not `main`, since milestones are relevant only for `main`.
+				// We add some sanity checks on the payload for extra precaution.
+				if (pr.base?.ref && typeof pr.base.ref === 'string' && pr.base.ref !== 'main') {
```

Comment:
We can remove these extra checks if we think they are too much—technically they should always be true.

Reply:
The only check I'd be unsure of is `pr.base.ref === 'string'`. Is it possible for this to not be a `string`?

Reply:
Doesn't seem so, according to the doc. Good to me to remove it!